crates/astria-composer/src/searcher/executor.rs

```diff
@@ -164,28 +216,43 @@ impl Executor {
                             break Err(e).wrap_err("failed submitting bundle to sequencer");
                         }
                     }
+
+                    block_timer.as_mut().reset(Instant::now() + Duration::from_millis(self.block_time_ms));
                 }
 
-                // receive new bundle for processing
-                Some(bundle) = self.new_bundles.recv(), if submission_fut.is_terminated() => {
-                    // TODO(https://github.com/astriaorg/astria/issues/476): Attach the hash of the
-                    // bundle to the span. Linked GH issue is for agreeing on a hash for `SignedTransaction`,
-                    // but both should be addressed.
-                    let span =  info_span!(
-                        "submit bundle",
-                        nonce.initial = nonce,
-                        bundle.len = bundle.len(),
+                Some(bundle) = bundle_factory.pop_complete(), if submission_fut.is_terminated() => {
+                    debug!(
+                        bundle_len=bundle.len(),
+                        "submitting bundle to sequencer"
                     );
-                    submission_fut = SubmitFut {
-                        client: self.sequencer_client.clone(),
-                        address: self.address,
-                        nonce,
-                        signing_key: self.sequencer_key.clone(),
-                        state: SubmitState::NotStarted,
-                        bundle,
+                    submission_fut = self.make_submission_fut(nonce, bundle);
+                }
+
+                // receive new seq_action and bundle it
+                Some(seq_action) = self.seq_actions_rx.recv(), if submission_fut.is_terminated() => {
```

Comment:
Should this be disabled if the submission future is in flight? Doesn't that defeat the purpose of having this as an extra branch?

Reply:
good catch